### PR TITLE
Adding struct checks to files. This solves countless of issues and bugs.

### PR DESCRIPTION
**Pull request by NeonKnightOA**

### CODE REVIEW

File: src/g_spawn.c [hunk 1]
```diff
@@ -456,7 +461,7 @@ char *ED_ParseEdict (char *data, edict_t *ent)
 	char		keyname[256];
 	const char	*com_token;
 
-	if (!ent || !data)
```
[the remark this answers is not in the document]
Comment (NeonKnightOA) [reply src/g_spawn.c]: Already solved.

File: src/g_phys.c [hunk 2]
```diff
@@ -175,21 +195,26 @@ int SV_FlyMove (edict_t *ent, float time, int mask)
...
-	
+
 	time_left = time;
 
 	ent->groundentity = NULL;
 	for (bumpcount=0 ; bumpcount<numbumps ; bumpcount++)
 	{
 		for (i=0 ; i<3 ; i++)
-			end[i] = ent->s.origin[i] + time_left * ent->velocity[i];
+		end[i] = ent->s.origin[i] + time_left * ent->velocity[i];
```
[the remark this answers is not in the document]
Comment (NeonKnightOA) [reply src/g_phys.c]: Indeed, I missed that tab. Solving...

File: src/g_phys.c [hunk 3]
```diff
@@ -255,12 +280,12 @@ int SV_FlyMove (edict_t *ent, float time, int mask)
 				if ((j != i) && !VectorCompare (planes[i], planes[j]))
 				{
 					if (DotProduct (new_velocity, planes[j]) < 0)
-						break;	// not ok
+					break;	// not ok
```
[the remark this answers is not in the document]
Comment (NeonKnightOA) [reply src/g_phys.c]: Indeed, I missed that tab. Solving...

File: src/g_target.c [hunk 4]
```diff
@@ -307,7 +382,7 @@ void use_target_changelevel (edict_t *self, edict_t *other, edict_t *activator)
 	if (deathmatch->value)
 	{
 		if (activator && activator->client)
-			gi.bprintf (PRINT_HIGH, "%s exited the level.\n", activator->client->pers.netname);
+		gi.bprintf (PRINT_HIGH, "%s exited the level.\n", activator->client->pers.netname);
```
[the remark this answers is not in the document]
Comment (NeonKnightOA) [reply src/g_target.c]: Indeed, I missed that tab. Solving...

File: src/g_target.c [hunk 5]
```diff
@@ -640,7 +795,7 @@ void target_laser_start (edict_t *self)
 		{
 			ent = G_Find (NULL, FOFS(targetname), self->target);
 			if (!ent)
-				gi.dprintf ("%s at %s: %s is a bad target\n", self->classname, vtos(self->s.origin), self->target);
+			gi.dprintf ("%s at %s: %s is a bad target\n", self->classname, vtos(self->s.origin), self->target);
```
[the remark this answers is not in the document]
Comment (NeonKnightOA) [reply src/g_target.c]: Indeed, I missed that tab. Solving...

File: src/zaero/ai.c [hunk 6]
```diff
@@ -100,57 +102,60 @@ int zFindRoamYaw(edict_t *self, float distcheck)
...
-    if(fabs(current - self->ideal_yaw) <= 359.0)
-    {
-      return 0;
-    }
-  }
+	float current = anglemod(self->s.angles[YAW]);
+
+	if (!self)
+	{
+		return 00;
```
[the remark this answers is not in the document]
Comment (NeonKnightOA) [reply src/zaero/ai.c]: Mistake. Fixing.

**Comment by NeonKnightOA** [comment]
Done. All found things solved.

**Comment by Yamagi** [comment]
Wow. Nice to see that finally someone puts some effort into Zaero. I never had the time nor the motivation for it and likely will never have it...

There're some compile errors left:
```===> Building game.so
gmake release/game.so
gmake[1]: Verzeichnis „/usr/home/yamagi/temp/yq2-zaero“ wird betreten
===> CC src/g_ai.c
===> CC src/g_cmds.c
src/g_cmds.c:11:3: error: non-void function 'ClientTeam' should return a value [-Wreturn-type]
                return;
                ^
1 error generated.
gmake[1]: [Makefile:162: build/src/g_cmds.o] Fehler 1 (ignoriert)
===> CC src/g_combat.c
src/g_combat.c:20:10: error: use of undeclared identifier 'qfalse'; did you mean 'false'?
                return qfalse;
                       ^~~~~~
                       false
src/header/shared.h:22:15: note: 'false' declared here
typedef enum {false, true}  qboolean;
              ^
src/g_combat.c:659:7: error: use of undeclared identifier 'targ'
        if (!targ || !inflictor || !attacker)
             ^
2 errors generated.
gmake[1]: [Makefile:162: build/src/g_combat.o] Fehler 1 (ignoriert)
===> CC src/g_func.c
src/g_func.c:1556:7: error: use of undeclared identifier 'ent'
        if (!ent || !other)
             ^
1 error generated.
gmake[1]: [Makefile:162: build/src/g_func.o] Fehler 1 (ignoriert)
===> CC src/g_items.c
===> CC src/g_main.c
===> CC src/g_misc.c
===> CC src/g_monster.c
===> CC src/g_phys.c
src/g_phys.c:477:10: error: returning 'void *' from a function with incompatible result type 'trace_t'
                return NULL;
                       ^~~~
/usr/include/sys/_null.h:34:14: note: expanded from macro 'NULL'
#define NULL    ((void *)0)
                ^~~~~~~~~~~
1 error generated.
gmake[1]: [Makefile:162: build/src/g_phys.o] Fehler 1 (ignoriert)
===> CC src/g_spawn.c
===> CC src/g_svcmds.c
===> CC src/g_target.c
===> CC src/g_trigger.c
===> CC src/g_turret.c
===> CC src/g_utils.c
===> CC src/g_weapon.c
===> CC src/monster/actor/actor.c
===> CC src/monster/berserker/berserker.c
===> CC src/monster/boss/boss.c
===> CC src/monster/boss2/boss2.c
src/monster/boss2/boss2.c:570:3: error: non-void function 'Boss2_CheckAttack' should return a value [-Wreturn-type]
                return;
                ^
1 error generated.
gmake[1]: [Makefile:162: build/src/monster/boss2/boss2.o] Fehler 1 (ignoriert)
===> CC src/monster/boss3/boss3.c
===> CC src/monster/boss3/boss31.c
===> CC src/monster/boss3/boss32.c
===> CC src/monster/brain/brain.c
===> CC src/monster/chick/chick.c
===> CC src/monster/flipper/flipper.c
===> CC src/monster/float/float.c
===> CC src/monster/flyer/flyer.c
===> CC src/monster/gladiator/gladiator.c
===> CC src/monster/gunner/gunner.c
===> CC src/monster/handler/handler.c
===> CC src/monster/hound/hound.c
===> CC src/monster/hover/hover.c
===> CC src/monster/infantry/infantry.c
===> CC src/monster/insane/insane.c
===> CC src/monster/medic/medic.c
===> CC src/monster/misc/move.c
===> CC src/monster/mutant/mutant.c
===> CC src/monster/parasite/parasite.c
===> CC src/monster/sentien/sentien.c
===> CC src/monster/soldier/soldier.c
===> CC src/monster/supertank/supertank.c
===> CC src/monster/tank/tank.c
===> CC src/player/client.c
src/player/client.c:173:3: error: non-void function 'IsFemale' should return a value [-Wreturn-type]
                return;
                ^
1 error generated.
gmake[1]: [Makefile:162: build/src/player/client.o] Fehler 1 (ignoriert)
===> CC src/player/hud.c
===> CC src/player/trail.c
===> CC src/player/view.c
===> CC src/player/weapon.c
===> CC src/savegame/savegame.c
===> CC src/shared/flash.c
===> CC src/shared/shared.c
===> CC src/zaero/acannon.c
===> CC src/zaero/ai.c
===> CC src/zaero/anim.c
===> CC src/zaero/camera.c
===> CC src/zaero/frames.c
===> CC src/zaero/item.c
===> CC src/zaero/mtest.c
===> CC src/zaero/spawn.c
===> CC src/zaero/trigger.c
===> CC src/zaero/weapon.c
===> LD release/game.so
yamagi@aka:pts/9 + ~/temp/yq2-zaero: gmake -i                                                                                                                                                                                       [16:03:50]
===> Building game.so
gmake release/game.so
gmake[1]: Verzeichnis „/usr/home/yamagi/temp/yq2-zaero“ wird betreten
===> CC src/g_ai.c
===> CC src/g_cmds.c
src/g_cmds.c:11:3: error: non-void function 'ClientTeam' should return a value [-Wreturn-type]
                return;
                ^
1 error generated.
gmake[1]: [Makefile:162: build/src/g_cmds.o] Fehler 1 (ignoriert)
===> CC src/g_combat.c
src/g_combat.c:20:10: error: use of undeclared identifier 'qfalse'; did you mean 'false'?
                return qfalse;
                       ^~~~~~
                       false
src/header/shared.h:22:15: note: 'false' declared here
typedef enum {false, true}  qboolean;
              ^
src/g_combat.c:659:7: error: use of undeclared identifier 'targ'
        if (!targ || !inflictor || !attacker)
             ^
2 errors generated.
gmake[1]: [Makefile:162: build/src/g_combat.o] Fehler 1 (ignoriert)
===> CC src/g_func.c
src/g_func.c:1556:7: error: use of undeclared identifier 'ent'
        if (!ent || !other)
             ^
1 error generated.
gmake[1]: [Makefile:162: build/src/g_func.o] Fehler 1 (ignoriert)
===> CC src/g_items.c
===> CC src/g_main.c
===> CC src/g_misc.c
===> CC src/g_monster.c
===> CC src/g_phys.c
src/g_phys.c:477:10: error: returning 'void *' from a function with incompatible result type 'trace_t'
                return NULL;
                       ^~~~
/usr/include/sys/_null.h:34:14: note: expanded from macro 'NULL'
#define NULL    ((void *)0)
                ^~~~~~~~~~~
1 error generated.
gmake[1]: [Makefile:162: build/src/g_phys.o] Fehler 1 (ignoriert)
===> CC src/g_spawn.c
===> CC src/g_svcmds.c
===> CC src/g_target.c
===> CC src/g_trigger.c
===> CC src/g_turret.c
===> CC src/g_utils.c
===> CC src/g_weapon.c
===> CC src/monster/actor/actor.c
===> CC src/monster/berserker/berserker.c
===> CC src/monster/boss/boss.c
===> CC src/monster/boss2/boss2.c
src/monster/boss2/boss2.c:570:3: error: non-void function 'Boss2_CheckAttack' should return a value [-Wreturn-type]
                return;
                ^
1 error generated.
gmake[1]: [Makefile:162: build/src/monster/boss2/boss2.o] Fehler 1 (ignoriert)
===> CC src/monster/boss3/boss3.c
===> CC src/monster/boss3/boss31.c
===> CC src/monster/boss3/boss32.c
===> CC src/monster/brain/brain.c
===> CC src/monster/chick/chick.c
===> CC src/monster/flipper/flipper.c
===> CC src/monster/float/float.c
===> CC src/monster/flyer/flyer.c
===> CC src/monster/gladiator/gladiator.c
===> CC src/monster/gunner/gunner.c
===> CC src/monster/handler/handler.c
===> CC src/monster/hound/hound.c
===> CC src/monster/hover/hover.c
===> CC src/monster/infantry/infantry.c
===> CC src/monster/insane/insane.c
===> CC src/monster/medic/medic.c
===> CC src/monster/misc/move.c
===> CC src/monster/mutant/mutant.c
===> CC src/monster/parasite/parasite.c
===> CC src/monster/sentien/sentien.c
===> CC src/monster/soldier/soldier.c
===> CC src/monster/supertank/supertank.c
===> CC src/monster/tank/tank.c
===> CC src/player/client.c
src/player/client.c:173:3: error: non-void function 'IsFemale' should return a value [-Wreturn-type]
                return;
                ^
1 error generated.
gmake[1]: [Makefile:162: build/src/player/client.o] Fehler 1 (ignoriert)
===> CC src/player/hud.c
===> CC src/player/trail.c
===> CC src/player/view.c
===> CC src/player/weapon.c
===> CC src/savegame/savegame.c
===> CC src/shared/flash.c
===> CC src/shared/shared.c
===> CC src/zaero/acannon.c
===> CC src/zaero/ai.c
===> CC src/zaero/anim.c
===> CC src/zaero/camera.c
===> CC src/zaero/frames.c
===> CC src/zaero/item.c
===> CC src/zaero/mtest.c
===> CC src/zaero/spawn.c
===> CC src/zaero/trigger.c
===> CC src/zaero/weapon.c
===> LD release/game.so
cc: error: no such file or directory: 'build/src/g_cmds.o'
cc: error: no such file or directory: 'build/src/g_combat.o'
cc: error: no such file or directory: 'build/src/g_func.o'
cc: error: no such file or directory: 'build/src/g_phys.o'
cc: error: no such file or directory: 'build/src/monster/boss2/boss2.o'
cc: error: no such file or directory: 'build/src/player/client.o'
gmake[1]: [Makefile:256: release/game.so] Fehler 1 (ignoriert)
gmake[1]: Verzeichnis „/usr/home/yamagi/temp/yq2-zaero“ wird verlassen
```

**Comment by NeonKnightOA** [comment]
Taking care of those.

**Comment by NeonKnightOA** [comment]
Done @Yamagi . I've also squashed all the commits in this PR.

**Comment by Yamagi** [comment]
Perfect. Thank you! :)

**Pull request merged**